### PR TITLE
Programming expressions -- seed and serialize non-Spritelab code docs

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Serialize new programming expressions fields for applab, gamelab, and weblab. Spritelab works a bit differently and will be addressed in a future PR. Also, this PR adds a call to serialize these objects after save in the controller.

Tested this by serializing all the progamming expressions locally, destroying them all, reseeding, then serializing again, then checking there were no diffs. I also added a unit test to this effect.

For review, I would recommend looking at the [diff of only .rb files](https://github.com/code-dot-org/code-dot-org/pull/44019/files?authenticity_token=5h9W%2BbNMRwohta9MfHAt2Vnnlz%2B%2FgXqeP1N9SPTQgYMuJwtV4tZjlbEKdKY7pdKMB8xCQNcIFqOTlnuTQwUOFA%3D%3D&file-filters%5B%5D=.rb&hide-deleted-files=true) then skimming the config changes.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/programming_expression.rb [hunk 1]
```diff
@@ -66,8 +66,7 @@ def self.properties_from_file(path, content)
 
     environment_name = File.basename(File.dirname(path)) == 'GamelabJr' ? 'spritelab' : File.basename(File.dirname(path))
     programming_environment = ProgrammingEnvironment.find_by(name: environment_name)
-
-    syntax = ProgrammingExpression.get_syntax(expression_config)
+    throw "Cannot find ProgrammingEnvironment #{environment_name}" unless programming_environment
```
Comment (davidsbailey) [inline dashboard/app/models/programming_expression.rb:69]: nit: can just use `find_by!` on the previous line if you want this behavior, unless you prefer this error message (I forget what the default one includes)

**Review by davidsbailey — APPROVED** [review]
LGTM, thanks for breaking this down!

**Review by dmcavoy — APPROVED** [review]
Excited to see this coming along!

File: dashboard/app/models/programming_expression.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/app/models/programming_expression.rb:69]: When I try this on the command line, I get
```
[development] dashboard > ProgrammingEnvironment.find_by!(name: 'blah')
  ProgrammingEnvironment Load (6.6ms)  SELECT  `programming_environments`.* FROM `programming_environments` WHERE `programming_environments`.`name` = 'blah' LIMIT 1
Traceback (most recent call last):
        1: from (irb):7
ActiveRecord::RecordNotFound (Couldn't find ProgrammingEnvironment)
```

I'd like to have the name of the missing environment in this case, so I'd rather keep it as is

Comment (davidsbailey) [reply dashboard/app/models/programming_expression.rb:69]: 👍  makes sense!

**Pull request merged**